rollout update strategy annotation

WnP opened the pull request:
Related to #591 and #724.

Since #724, the `Rollout` update strategy has been changed from "rollout" to "restart". While I can understand that this may be useful in some scenarios, it may not be expected in others.

This MR gives the user the flexibility to choose between a "rollout" or "restart" strategy using an annotation.

MuneebAijaz commented:
https://github.com/stakater/Reloader/pull/771/files there is a PR already, we were waiting for tests to be added. If you can pursue/add tests via your PR, we can merge yours and close the other one.

WnP commented:
Hi @MuneebAijaz,

Of course, I've updated my PR with some unit tests.

During the process I've changed `testutil.getObjectMeta` to be able to pass some annotations.

I've also removed some log lines:
```
-       logrus.Warnf("Before: %+v", rolloutBefore.Spec.Template.Spec.Containers[0].Env)
-       logrus.Warnf("After: %+v", rollout.Spec.Template.Spec.Containers[0].Env)
```
And the api call to retrieve `rolloutBefore`.
